pkg/ruler/storage/cleaner/config.go

```diff
@@ -14,7 +15,7 @@ type Config struct {
 	Period time.Duration `yaml:"period,omitempty"`
 }
 
-func (c Config) RegisterFlags(f *flag.FlagSet) {
+func (c *Config) RegisterFlags(f *flag.FlagSet) {
 	f.DurationVar(&c.MinAge, "ruler.wal-cleaner.min-age", DefaultCleanupAge, "The minimum age of a WAL to consider for cleaning.")
-	f.DurationVar(&c.Period, "ruler.wal-cleaer.period", DefaultCleanupPeriod, "How often to run the WAL cleaner.")
+	f.DurationVar(&c.Period, "ruler.wal-cleaner.period", DefaultCleanupPeriod, "How often to run the WAL cleaner. 0 = disabled.")
```

Comment:
There was a typo on this CLI flag. What is the best course of action here? Should we keep this flag (`ruler.wal-cleaer.period`) and deprecate it and create a new one (`ruler.wal-cleaner.period`)?